packages/cli/src/controller/codegen-controller.ts

```diff
@@ -306,9 +306,11 @@ export function processAbis(
   return renderInterfaceJobs;
 }
 
-function joinInputAbiName(abiObject: abiInterface) {
+export function joinInputAbiName(abiObject: abiInterface) {
   // example: "TextChanged_bytes32_string_string_string_Event", Event name/Function type name will be joined in ejs
-  const inputToSnake: string = abiObject.inputs.map((obj) => obj.type.toLowerCase()).join('_');
+  const inputToSnake = abiObject.inputs
+    .map((obj) => (obj.type.includes('[]') ? obj.type.replace(/\[\]/g, '_arr') : obj.type.toLowerCase()))
```

Comment:
The includes check is redundant. If there's no match it wont replace and we should lowercase even if there's an `[]`